cli/azd/pkg/project/service_config.go

```diff
@@ -155,79 +156,7 @@ func (sc *ServiceConfig) GetFrameworkService(
 		frameworkService = NewDockerProject(sc, env, docker.NewDocker(commandRunner), sourceFramework)
 	}
 
-	return &frameworkService, nil
-}
-
-// Adds an event handler for the specified event name
-func (sc *ServiceConfig) AddHandler(name Event, handler ServiceLifecycleEventHandlerFn) error {
-	newHandler := fmt.Sprintf("%v", handler)
-	events := sc.handlers[name]
-
-	for _, ref := range events {
-		existingHandler := fmt.Sprintf("%v", ref)
-
-		if newHandler == existingHandler {
-			return fmt.Errorf("event handler has already been registered for %s event", name)
-		}
-	}
-
-	events = append(events, handler)
-	sc.handlers[name] = events
-
-	return nil
-}
```

Comment:
Common event handling structure was moved out out to `ext.EventDispatcher` and now using struct embedding to enable this feature on both `project.ProjectConfig` and `project.ServiceConfig`